File: backend/Models/CustomFieldValue.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace backend.Models;

public class CustomFieldValue
{
    [Key]
    public Guid Id { get; set; }

    public Guid CustomerId { get; set; }
    [JsonIgnore]
    public Customer? Customer { get; set; }

    public Guid CustomFieldId { get; set; }
    public CustomField? CustomField { get; set; }

    public string? Value { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/CustomFieldValue.ts
export type TCustomFieldValue = {
  id: string;
  customerId: string;
  customFieldId: string;
  customField?: string;
  value: string | boolean;
  createdAt: string;
  updatedAt: string;
};

export type TAddCustomFieldValue = {
  customerId?: string;
  customFieldId: string;
  value: string | boolean;
};

export type TUpdateCustomFieldValue = {
  id: string;
  value: string | boolean;
};
